src/main/java/edu/umich/its/cpm/MigrationTaskService.java

```diff
@@ -869,7 +869,22 @@ private StringBuffer preMigrationChecks(StringBuffer itemStatus,
 				}
 				
 				// create box folder
-				BoxFolder parentFolder = new BoxFolder(api, boxFolderIdStack.peek());
+				String peekFolderId = "";
+				try
+				{
+					peekFolderId = boxFolderIdStack.peek();
```

Comment:
How about explicitly checking for an empty stack rather than picking up the exception?  That would be the only reason to trigger the exception and it is clearer.  A Java stack should have a 'size' (or 'length') method or an 'empty' predicate.  A comment on how this might happen would be good.